feat: support tachyon halo2

chokobole opened the pull request:
# Description

This PR enables producing a proof by Tachyon.

Commit:
feat(halo2_proofs): support constraint system serialization

Commit:
feat(halo2_proofs): revert 62a4421

Instead of replacing with MSM, the next commit tries to replace with
whole Halo2 CPU code. This is a change for better gpu utilization ratio
in the future.

Commit:
build: update rust version 1.63.0 to 1.70.0

This is needed because a package `home v0.5.9` requires rustc 1.70.0 or
newer.

Commit:
feat(halo2_proofs): make `column_type` to public

This is needed by tachyon versioned `create_proof()`.

Commit:
feat(halo2_proofs): add `create_proof()` of tachyon

fakedev9999 commented on halo2_proofs/src/plonk/tachyon.rs:
```diff
@@ -0,0 +1,441 @@
+    CurveAffine,
+};
+
+/// This creates a proof for the provided `circuit` when given the public
```
```suggestion
/// This creates a proof for the provided `circuits` when given the public
```

chokobole replied:
Actually, since I just took this comment from Halo2, I'm worried his may produce another diffs !

fakedev9999 commented on halo2_proofs/src/plonk/tachyon.rs:
```diff
@@ -0,0 +1,441 @@
+
+                // Add blinding factors to advice columns
+                for advice_values in &mut advice_values {
+                    //for cell in &mut advice_values[unusable_rows_start..] {
+                    //*cell = C::Scalar::random(&mut rng);
+                    //*cell = C::Scalar::one();
+                    //}
```
```suggestion
```

fakedev9999 commented on halo2_proofs/src/plonk/prover.rs:
```diff
@@ -502,11 +507,13 @@ pub fn create_proof<
     // Obtain challenge for keeping all separate gates linearly independent
     let y: ChallengeY<_> = transcript.squeeze_challenge_scalar();
     println!("Halo2(y): {:?}", *y);
-
```
```suggetion

```

dongchangYoo committed:
feat(halo2_proofs): add logging at each level

- `RUST_LOG=info`: the benchmark log is outputted.
- `RUST_LOG=debug`: logs are outputted including squeezed values.
- `RUST_LOG=trace`: VKeyStr and all Proofs are outputted.

Insun35 reviewed (approved):
LGTM

dongchangYoo reviewed (approved):
LGTM

fakedev9999 reviewed (approved):
LGTM
